### PR TITLE
JAV-579 support dataStrArray

### PR DESCRIPTION
**Pull request by mt-monster**

### CODE REVIEW

File: common/common-rest/src/test/java/io/servicecomb/common/rest/codec/param/TestCsvQueryProcessor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package io.servicecomb.common.rest.codec.param;
```
Comment (WillemJiang) [inline common/common-rest/src/test/java/io/servicecomb/common/rest/codec/param/TestCsvQueryProcessor.java]: Please add the license header

File: demo/demo-jaxrs/jaxrs-server/src/main/java/io/servicecomb/demo/jaxrs/server/CodeFirstJaxrs.java [hunk 2]
```diff
@@ -202,4 +206,13 @@ public String testRawJsonAnnotation(@RawJsonRequestBody String jsonInput) {
     }
     return "hello " + person.get("name");
   }
+
+  @Path("/testGetStrArray")
+  @GET
+  public String[] testGetStrArray(@RequestParamColFmt(collectionFormat = "csv") String[] str) {
+    for (int i = 0; i < str.length; i++) {
+      System.out.println("*******"+str[i]);
```
Comment (WillemJiang) [inline demo/demo-jaxrs/jaxrs-server/src/main/java/io/servicecomb/demo/jaxrs/server/CodeFirstJaxrs.java]: It's better to use to logger,  so we can decide to print it to console or not.

File: demo/demo-springmvc/springmvc-server/src/main/java/io/servicecomb/demo/springmvc/server/CodeFirstSpringmvc.java [hunk 3]
```diff
@@ -343,4 +344,12 @@ public String metricsForTest() {
       throw new InvocationException(500, "500", "JsonProcessingException", e);
     }
   }
+
+  @RequestMapping(path = "/testGetStrArray", method = RequestMethod.GET)
+  public String[] testGetStrArray(@RequestParamColFmt(collectionFormat = "csv") String[] str) {
+    for (int i = 0; i < str.length; i++) {
+      System.out.println("*******"+str[i]);
```
Comment (WillemJiang) [inline demo/demo-springmvc/springmvc-server/src/main/java/io/servicecomb/demo/springmvc/server/CodeFirstSpringmvc.java]: Same suggestion here.

File: swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/generator/core/processor/parametertype/RequestParamColFmtAnnotationProcessor.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package io.servicecomb.swagger.generator.core.processor.parametertype;
```
Comment (WillemJiang) [inline swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/generator/core/processor/parametertype/RequestParamColFmtAnnotationProcessor.java]: Please add the license header.

Comment (mt-monster) [reply swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/generator/core/processor/parametertype/RequestParamColFmtAnnotationProcessor.java]: already done

File: common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/QueryProcessorCreator.java [hunk 5]
```diff
@@ -26,10 +29,34 @@
...
+
+  @Override
+  public ParamValueProcessor create(Parameter parameter, Type genericParamType) {
+    JavaType targetType = TypeFactory.defaultInstance().constructType(genericParamType);
+
+    String collectionFormat = ((QueryParameter) parameter).getCollectionFormat();
+
+    if (collectionFormat == null) {
+      return new QueryProcessor(parameter.getName(), targetType);
+    } else if (collectionFormat.equals("csv")) {
```
Comment (wujimin) [inline common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/QueryProcessorCreator.java]: no need to write "else"

File: swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/extend/annotations/RequestParamColFmt.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.core.annotation.AliasFor;
+
+@Target({ElementType.PARAMETER})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface RequestParamColFmt {
```
Comment (wujimin) [inline swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/extend/annotations/RequestParamColFmt.java]: do  not  create a new annotation.
use  swagger annotation

Comment (mt-monster) [reply swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/extend/annotations/RequestParamColFmt.java]: use swagger annotation is diffcult to solve this problem .beacause 1: queryparameter has fixed default collection value ,it means should extend it to cover this method to return collection value.2:
no matter i choose ApiParam or ApiImplicitParam ,ApiParam index cannot be fixed ,so cannot process, ApiImplicitParam dataType has no queryType,so cannot process

Comment (wujimin) [reply swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/extend/annotations/RequestParamColFmt.java]: we have discuss this weeks age, and i have gave you the solution......

File: common/common-rest/src/test/java/io/servicecomb/common/rest/codec/param/TestCsvQueryProcessor.java (hunk 1 above)
Comment (WillemJiang) [reply common/common-rest/src/test/java/io/servicecomb/common/rest/codec/param/TestCsvQueryProcessor.java]: The license header is still missing.

**Pull request closed without merging**